packages/menu/src/Menu.tsx

```diff
@@ -95,51 +98,31 @@ function Menu({
   trigger,
   ...rest
 }: MenuProps) {
-  const refs: Array<HTMLElement> = [];
-  const titleArr: Array<string> = [];
-
   const hasSetInitialFocus = useRef(false);
   const hasSetInitialOpen = useRef(false);
-  const [focused, setFocused] = useState<HTMLElement>(refs[0] || null);
 
   const [, setClosed] = useState(false);
-  const [currentSubMenu, setCurrentSubMenu] = useState<React.ReactElement<
-    HTMLLIElement
-  > | null>(null);
+  const [currentSubMenu, setCurrentSubMenu] = useState<SubMenuElement | null>(
+    null,
+  );
   const [uncontrolledOpen, uncontrolledSetOpen] = useState(false);
 
   const setOpen =
     (typeof controlledOpen === 'boolean' && controlledSetOpen) ||
     uncontrolledSetOpen;
   const open = controlledOpen ?? uncontrolledOpen;
 
-  // When a SubMenu becomes open, it's set to currentSubMenu, and we focus on the first child.
-  const focusedRefIndex = refs.indexOf(focused);
-
-  let item: HTMLElement | undefined;
-
-  if (focusedRefIndex !== -1) {
-    if (!currentSubMenu) {
-      const subMenu = refs[focusedRefIndex];
-      item = subMenu;
-    } else {
-      const subMenuFirstChild = refs[focusedRefIndex + 1];
-      item = subMenuFirstChild;
-    }
-  }
-
-  useEffect(() => {
-    item?.focus();
-  }, [item]);
-
-  const updatedChildren = React.useMemo(() => {
+  const { updatedChildren, refs } = React.useMemo(() => {
```

Comment:
This PR is reminding me that I should take a look at focus-handling in the Menu component. I think we misunderstood some guidelines, and we can probably make this entire component dramatically simpler. (out of scope here, but more a note to self 🤔 )

Reply:
That said, these changes do require a changeset before merging in